pkg/ingester/activeseries/active_series.go

```diff
@@ -114,14 +114,14 @@ func (c *ActiveSeries) ConfigDiffers(ctCfg asmodel.CustomTrackersConfig, caCfg *
 	return ctCfg.String() != c.matchers.Config().String() || caCfg != c.cat
 }
 
-func (c *ActiveSeries) ReloadMatchers(asm *asmodel.Matchers, now time.Time) {
+func (c *ActiveSeries) ReloadMatchersAndTrackers(asm *asmodel.Matchers, cat *costattribution.ActiveSeriesTracker, now time.Time) {
 	c.configMutex.Lock()
 	defer c.configMutex.Unlock()
-
 	for i := 0; i < numStripes; i++ {
-		c.stripes[i].reinitialize(asm, &c.deleted, c.cat)
+		c.stripes[i].reinitialize(asm, &c.deleted, cat)
 	}
 	c.matchers = asm
+	c.cat = cat
```

Comment:
Here is the bug, it occurred because c.cat was compared with the current cost attribution tracker when the config changed, but it was never updated beforehand. As a result, active_series got stuck during the active series loading process.